Monika-After-Story/MonikaModDev
# Gift hints

closes #5412 

Adds a topic to explain the gifting mechanic via a note to the player.

Also modifies the poem framework slightly to allow chibika notes with alt fonts using the `mas_generic_poem` screen

## Testing:
- Verify that on persists which have gifted before, the topic is not random and never shows up
- Verify that on a fresh persistent, when the exp level reaches at least 15 and having not gifted anything yet, the topic is queued
- Verify that if the topic hasn't run and it is a special day, the topic is queued
- Verify that the topic doesn't unlock

### Other testing:
- Verify that the corrupted persistent (open persistent in notepad and add characters to corrupt) topic still works correctly (both paths)
- Verify that poems still work normally
